### DISCUSSION TITLE
Support getting directions via openrouteservice

### DISCUSSION THREAD

**Discussion by ParetoOptimalDev** [minad/osm#42] [Ideas]
Today I wanted to use osm.el to get directions from one place to another, but realized there wasn't support.

I realized why and searched to see if there were open source solutions and there is openrouteservice, a free directions REST API compatible with OpenStreetMap which crowd sources directions.

Here is the API:

https://openrouteservice.org/dev/#/api-docs/v2/directions

I'm not sure if you are interested or have a need, but it seems like a fun/useful addition :smile:

**Comment by minad** [comment]
Yes, this would be nice to have. Route drawing is already supported, but there is no API support yet. Patches welcome.

**Comment by minad** [comment]
Another good alternative seems to be: https://routing.openstreetmap.de/

**Reply by minad** [reply]
Implemented in https://github.com/minad/osm/commit/7e78000a7a2937f479c25f6c1e97d838f16104b5
